clients/azuredevopsrepo/commits.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2024 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuredevopsrepo
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/git"
+	"github.com/ossf/scorecard/v5/clients"
+)
+
+type commitsHandler struct {
+	gitClient           git.Client
+	once                *sync.Once
+	errSetup            error
+	repourl             *Repo
+	commitsRaw          *[]git.GitCommitRef
+	pullRequestsRaw     *git.GitPullRequestQuery
+	commitDepth         int
+	getCommits          fnGetCommits
+	getPullRequestQuery fnGetPullRequestQuery
+}
+
+func (handler *commitsHandler) init(repourl *Repo, commitDepth int) {
+	handler.repourl = repourl
+	handler.errSetup = nil
+	handler.once = new(sync.Once)
+	handler.commitDepth = commitDepth
+	handler.getCommits = handler.gitClient.GetCommits
+	handler.getPullRequestQuery = handler.gitClient.GetPullRequestQuery
+}
+
+type (
+	fnGetCommits          func(ctx context.Context, args git.GetCommitsArgs) (*[]git.GitCommitRef, error)
+	fnGetPullRequestQuery func(ctx context.Context, args git.GetPullRequestQueryArgs) (*git.GitPullRequestQuery, error)
+)
+
+func (handler *commitsHandler) setup() error {
+	handler.once.Do(func() {
+		opt := git.GetCommitsArgs{
+			RepositoryId: &handler.repourl.ID,
+			Top:          &handler.commitDepth,
+			SearchCriteria: &git.GitQueryCommitsCriteria{
+				ItemVersion: &git.GitVersionDescriptor{
+					VersionType: &git.GitVersionTypeValues.Branch,
+					Version:     &handler.repourl.defaultBranch,
```

Comment:
note: Scorecard also has a `--commit` argument, which you have stored in the `handler.repourl.commitSHA`. So we should respect that